lib/pure/rationals.nim

```diff
@@ -122,30 +122,37 @@ func toInt*[T](x: Rational[T]): int =
 
 func `+`*[T](x, y: Rational[T]): Rational[T] =
   ## Adds two rational numbers.
+  if x.num == 0: return y
+  if y.num == 0: return x
   let common = lcm(x.den, y.den)
   result.num = common div x.den * x.num + common div y.den * y.num
   result.den = common
   reduce(result)
 
 func `+`*[T](x: Rational[T], y: T): Rational[T] =
   ## Adds the rational `x` to the int `y`.
+  if x.num == 0: return toRational y
```

Comment:
```suggestion
  if x.num == 0: return toRational(y)
```
Calling functions with parentheses is preferred most of the time afaik.

Reply:
I naturally want to follow the stdlib style. However, I wasn't able to find this particular thing. Do you know if this is documented somewhere?
Personally the lack of visual clutter/noise is one of the things why Nim is appealing to me. Unnecessary parenthesis (those that do not carry a semantic information) _is_ visual clutter. To me, function chaining (either left to right using `.` or  right to left using ` `) is more readable.

Reply:
Agreed.